common/retry_config.go

```
@@ -0,0 +1,115 @@
+package common
+
+import (
+	"fmt"
+
+	"github.com/agglayer/aggkit/config/types"
+)
+
+type RetryConfigMode string
+
+const (
+	RetryConfigModeNoRetries RetryConfigMode = ""
+	RetryConfigModeDelays    RetryConfigMode = "delays"
+	RetryConfigModeBackoff   RetryConfigMode = "backoff"
+)
+
+var (
+	ErrInvalidRetryConfigMode                       = fmt.Errorf("invalid retry config mode")
+	_                         RetryPolicyConfigurer = (*RetryPolicyGenericConfig)(nil)
+)
+
+// RetryPolicyGenericConfig defines the configuration for retry policies in the system.
+// it's a merge of struct RetryBackoffConfig and RetryDelaysConfig in order of
+// simplify reading from config file (check types)
+type RetryPolicyGenericConfig struct {
+	Mode              RetryConfigMode // "delays", "backoff", or "" for no retries
+	MaxRetries        int
+	Delays            []types.Duration
+	InitialBackoff    types.Duration
+	MaxBackoff        types.Duration
+	BackoffMultiplier float64
+	// internal field to cache the underlaying config
+	cache RetryPolicyConfigurer `mapstructure:"-"`
+}
+
+// RetryPolicyConfigurer is an interface that defines methods for configuring retry policies.
+// Each class that implements a retry policy configuration should implement this interface.
+type RetryPolicyConfigurer interface {
+	// Validate configuration
+	Validate() error
+	// NewRetryHandler returns a RetryHandler based on the configuration
+	NewRetryHandler() *RetryHandler
+	// String returns a string representation of the configuration
+	String() string
+	// Brief is a brief string representation of the object
+	Brief() string
+}
+
+func (r *RetryPolicyGenericConfig) Validate() error {
+	cfg, err := r.Factory()
+	if err != nil {
+		return err
+	}
+	return cfg.Validate()
+}
+
+func (r *RetryPolicyGenericConfig) NewRetryHandler() *RetryHandler {
+	cfg, err := r.Factory()
+	if err != nil {
+		return NewRetryHandler(nil, 0) // return a no-op handler
+	}
```

Let's not silence this error, we should at least log it (if not interrupt the booting of the aggkit altogether). WDYT?